src/structs.rs

```diff
@@ -107,6 +107,63 @@ impl From<&AssetCode> for BaseField {
     }
 }
 
+/// Amount value type
+#[tagged_blob("AMOUNT_VALUE")]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Default,
+    Hash,
+    Eq,
+    From,
+    Into,
+    Add,
```

Comment:
What exactly does this derive? It would be useful to have `impl<T: Into<AmountValue>> Add<T> for AmountValue`, then with `From<u64>` e.g. we can write `amount_value + 1u64` rather than `amount_value + 1u64.into()`

Same comment for the other num traits.

Reply:
> What exactly does this derive?

i think `derive_more` would derive `From/To` for all primitive numeric types, if i read [their doc](https://jeltef.github.io/derive_more/derive_more/from.html#example-usage) right.

Reply:
I tried `Amount(1u64)` and `Amount::from(1u64)`, it doesn't compile unfortunately... I guess we have to impl From<u64> ourselves...

Reply:
Oh the key is to use `#[from(forward)]`

Reply:
Derive From/Into for all unsigned type, done in 0b1cd8a6c7526012a5585afc99eb6259ba7b62e4